### PR TITLE
Check if on secure desktop when checking if Windows is locked

### LINKED ISSUE #14286: Many scripts not available on secure screen
### Steps to reproduce:
You should have already configured NVDA to be used on Windows secure screen.
1. Start the computer to reach the logon screen
2. Press NVDA+N
3. Press NVDA+1
4. etc.
### Actual behavior:
2. No NVDA menu
3. Help mode is not enabled
### Expected behavior:
2. NVDA menu should open.
3. Help mode should be enabled
### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2022.4beta2
#### Windows version:
Windows 10 2004 (x64) build 19041.388
But same issue seen on other versions of Windows 10.
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Not tried.
But I remember that the issue was not occurring before. I think it has arrived with NVDA2022.4beta2.
#### If NVDA add-ons are disabled, is your problem still occurring?
Not tested.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Fixes #14286 

### Summary of the issue:
`isWindowsLocked` is meant to return `False` when on secure screens.
`isWindowsLocked` should return True, only when Windows is on the lock screen, to prevent users from escaping the lock screen.
This is because NVDA secure mode prevents a different set of exploits for secure screens.
2022.3.1 introduced a regression where when using a secure screen, `isWindowsLocked` returned True.

### Description of user facing changes
NVDA menu and other functionality is re-enabled on secure screens.

### Description of development approach
A new function has been added to check if NVDA is running on the secure desktop.
If that is the case, `isWindowsLocked` returns `False` ,

Returning `False` if NVDA is in a secure mode was not a viable path.
This is because if NVDA is running on secure mode via the `--secure` parameter, and a user navigates to the Windows lock screen, then there would be no security parameters.

### Testing strategy:
Test using NVDA on the sign-in screen, lock screen and UAC screen.

### Known issues with pull request:
None

### Change log entries:
Bug fixes
```
- Fixes a regression from 2022.3.1 where certain functionality was disabled on secure screens. (#14286)
```

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Review by feerrenrut — DISMISSED** [review]
Generally, looks good. I have some suggestions to help future developers.

It would be helpful to document the difference between `isSecureDesktop` and `globalVars.appArgs.secure`. I'd suggest doing this:
- In the `isSecureDesktop` definition.
- Where `secure` is added to the arg parser.
- In `globalVars`


Is [`serviceDebug` key](https://github.com/nvaccess/nvda/blob/2fd889937bc57ff65827050b3de7dd9cafb952d0/source/nvda.pyw#L348) documented? Looks like it would have been helpful. Assuming it makes NVDA on the system profile easier to debug, it would be good to have the effects documented and linked to from the "self signing NVDA" doc (since the primary reason we expect devs to do that is to test/debug on the system profile)

**Comment by seanbudd** [comment]
>Is [serviceDebug key](https://github.com/nvaccess/nvda/blob/2fd889937bc57ff65827050b3de7dd9cafb952d0/source/nvda.pyw#L348) documented? Looks like it would have been helpful. Assuming it makes NVDA on the system profile easier to debug, it would be good to have the effects documented and linked to from the "self signing NVDA" doc (since the primary reason we expect devs to do that is to test/debug on the system profile)


Yes, we have it documented in the [user guide](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#SystemWideParameters) and (technical design overview](https://github.com/nvaccess/nvda/blob/9b538f4d6e21bda08bc34998f2a657eb5a28eee5/devDocs/technicalDesignOverview.md). The serviceDebug parameter disables secure mode on secure screens, so it is not helpful when testing in this scenario.

File: source/systemUtils.py [hunk 1]
```diff
@@ -166,4 +166,11 @@ def _getDesktopName() -> str:
 
 
 def _isSecureDesktop() -> bool:
+	"""
+	When NVDA is running on a secure screen,
+	it is running on the secure desktop.
+	When the serviceDebug parameter is not set,
```
Comment (feerrenrut) [inline source/systemUtils.py:172]:
```suggestion
	When the serviceDebug parameter (see user_guide SystemWideParameters and devDocs/technicalDesignOverview.md 'Logging in secure mode') is not set,
```

File: source/globalVars.py [hunk 2]
```diff
@@ -32,6 +32,13 @@ class DefaultAppArgs(argparse.Namespace):
 	language: str = "en"
 	minimal: bool = False
 	secure: bool = False
+	"""
+	When this is True, NVDA is running in secure mode.
+	This is set to True when NVDA starts with the --secure parameter.
+	This is also set to True when NVDA is running on a secure screen
+	(systemUtils._isSecureDesktop() returns True)
+	and the serviceDebug parameter has not been set.
```
Comment (feerrenrut) [inline source/globalVars.py:40]:
```suggestion
	and the serviceDebug parameter (see user_guide SystemWideParameters and devDocs/technicalDesignOverview.md 'Logging in secure mode') is not set,
```

**Review by feerrenrut — DISMISSED** [review]

**Pull request merged**